python/taichi/lang/impl.py

```diff
@@ -454,6 +454,15 @@ def fused_string(entries):
     taichi_lang_core.create_print(contentries)
 
 
+@taichi_scope
+def ti_assert(cond, msg, extra_args):
+    # Mostly a wrapper to help us convert from ti.Expr (defined in Python) to
+    # taichi_lang_core.Expr (defined in C++)
+    import taichi as ti
+    taichi_lang_core.create_assert_stmt(
+        ti.Expr(cond).ptr, msg, [ti.Expr(x).ptr for x in extra_args])
```

Comment:
What if `extra_args` is `Vector`? Let's make use of `Vector.__ti_repr__` like currently `ti_print` does.

Reply:
I see. Unfortunately, assert format specifier doesn't support such a fancy data structure yet, see https://github.com/taichi-dev/taichi/blob/c3f1a5d1899500eb08facca78fa1373e279cb692/taichi/program/program.cpp#L445-L454

It only has `%d` or `%f`..